cinema/views.py

```diff
@@ -1 +1,51 @@
-# write views here
+from rest_framework import viewsets
+
+from cinema.models import Actor, Genre, CinemaHall, Movie, MovieSession
+from cinema.serializers import (
+    MovieListSerializer,
+    MovieSerializer,
+    MovieDetailSerializer,
+    MovieSessionSerializer,
+    MovieSessionListSerializer,
+    MovieSessionDetailSerializer,
+    ActorSerializer,
+    GenreSerializer,
+    CinemaHallSerializer
+)
+
+
+class GenreViewSet(viewsets.ModelViewSet):
+    queryset = Genre.objects.all()
+    serializer_class = GenreSerializer
+
+
+class ActorViewSet(viewsets.ModelViewSet):
+    queryset = Actor.objects.all()
+    serializer_class = ActorSerializer
+
+
+class CinemaHallViewSet(viewsets.ModelViewSet):
+    queryset = CinemaHall.objects.all()
+    serializer_class = CinemaHallSerializer
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.prefetch_related("genres", "actors")
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+        elif self.action == "retrieve":
+            return MovieDetailSerializer
+        return MovieSerializer
+
+
+class MovieSessionViewSet(viewsets.ModelViewSet):
+    queryset = MovieSession.objects.prefetch_related("cinema_hall")
```

Comment:
does it has m2m relations with cinema_hall? also what about movie ?
also the same for optimized query as above